src/stepFunctions/asl/aslServer.ts

```diff
@@ -266,6 +282,15 @@ function triggerValidation(textDocument: TextDocument): void {
     }, validationDelayMs)
 }
 
+// sets language service depending on document language
+function setLanguageService(textDocument: TextDocument): void {
+    if (textDocument.languageId == 'yasl') {
+        languageService = aslYamlLanguageService
+    } else {
+        languageService = aslJsonLanguageService
+    }
+}
```

Comment:
This function should probably return something instead of mutating some sort of global state.

Additionally, can be shortened to `textDocument.languageId == 'yasl' ? aslYamlLanguageService : aslJsonLanguageService`

Additionally, please use [`===` instead of `==`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Equality_comparisons_and_sameness)! This is the first I've seen of this but it definitely applies everywhere.

Reply:
Yep, I'll convert it into a getter function